src/chatwindow/chatwindowdock.cpp

```diff
@@ -1,13 +1,62 @@
 #include "chatwindowdock.h"
 
-#include <QException> // temp
+#include <QListWidgetItem>
+#include <QException>
+#include <QKeyEvent>
+#include <QDebug>
 
 #include "mainwindow.h"
 #include "chatwidget.h"
 
 ChatWindowDock::ChatWindowDock(QWidget *pParent): QDockWidget (pParent)
 {
     setWindowTitle("Chat");
-    pChatWidget = new ChatWidget;
-    setWidget(pChatWidget);
+    mpChatWidget = new ChatWidget;
+    connect(mpChatWidget, &ChatWidget::userToConnectSelected,
+            this, &ChatWindowDock::onUserToConnectSelected);
+    connect(mpChatWidget, &ChatWidget::sendMessage,
+            this, &ChatWindowDock::onSendMessage);
+
+    setWidget(mpChatWidget);
+    setMinimumWidth(200);
+}
+
+void ChatWindowDock::setUserName(const QString &userName)
+{
+    mpChatWidget->setCurrentUserName(userName);
+}
+
+void ChatWindowDock::displayMessage(const QString userName, const QString message)
+{
+    mpChatWidget->displayMessage(userName, message);
+}
+
+void ChatWindowDock::onUserToConnectSelected(QString userName)
+{
+    qDebug() << "dock slot";
```

Comment:
better to remove debug info

Reply:
All qDebugs in the chat classes are now removed

Reply:
removed